Gemfile

```diff
@@ -188,6 +184,7 @@ group :development, :test do
 end
 
 group :test do
+  gem 'puma' # required for tests which involve websockets
```

Comment:
is this solved with a newer version of `unicorn` ? probably not, but had to ask...

Reply:
Okay, after spending a bit more time on this, it turns out that `puma` is configured for Capybara [by default](https://github.com/teamcapybara/capybara/blob/2.15.1/lib/capybara.rb#L451-L454). `:unicorn` isn't in that file so you need to add it manually.

I tried removing `puma` from the Gemfile and sticking the following in `rails_helper` but it made the tests fail (`rspec spec/features/issue_pages/show_page_spec.rb` seem like the most relevant ones to try seeing as they're the ones which actually use ActionCable):

```ruby
Capybara.register_server :unicorn do |app, port, host, options={}|
  Rack::Handler::Unicorn.run(app, {Host: host, Port: port, Threads: "0:4", workers: 0, daemon: false}.merge(options))
end
Capybara.server = :unicorn
```

I'm probably passing the wrong settings to `Unicorn.run` (the above are just copied from what Capybara passes to `puma`) but I have no idea what to try instead.

Things I've tried that didn't work:

1. Updating `unicorn` to the latest version (5.4.0)
2. Updating `capybara` to the latest version (3.3.1)
3. Copying `config/unicorn.rb` from Pro to CE and passing it as an option (`config_file: ...`) to `Unicorn.run` (not surprised that this didn't work as that file is meant to set Unicorn up for a production environment.

I feel very uneasy using a different server for the production environment vs. the testing environment (although note that this is what we were doing before anyway, since production used unicorn while dev/test used WEBrick), especially since the difference between `puma` and `unicorn` can apparently have such a big effect on whether tests pass. But I don't know how we can fix this and I'm out my depth. I give up for now.

(Of course the other option is to switch to `puma` in dev/prod, as opposed to using `unicorn` in tests. I don't understand enough about the difference between the two servers to have a preference for which we use. The more important point is that we can't continue to use WEBrick in tests, or anywhere for that matter, if we want to start using ActionCable.)

Reply:
> although note that this is what we were doing before anyway, since production used unicorn while dev/test used WEBrick

from this perspective, it does not seem that terrible if we use `puma`in dev/test and skip this by fix by now

Reply:
Yeah, I think we can get away with it for now. Just wanted to give some more details about what I've tried as they may be helpful when trying to fix it in the future.